File: C#/Easy/9. Palindrome Number.cs
// https://leetcode.com/problems/palindrome-number

public class Solution {
    // Comments:
    // - x: reads the same forward and backwards
    public bool IsPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        long reverse = 0;
        int xx = x;

        while (x > 0) {
            reverse = (reverse * 10) + (x % 10);
            x /= 10;
        }

        return reverse == xx;
    }
}

File: C/Easy/9. Palindrome Number.c
// https://leetcode.com/problems/palindrome-number/description/

bool isPalindrome(int x) {

    char temp[20] = {0};
    sprintf(temp, "%d", x);

    int len = strlen(temp);
    char rev[20]; // increased size to 20 incase of overflow

    for (int i = 0; i < len; i++) {
        rev[i] = temp[len - i - 1];
    }

    rev[len] = '\0';

    if (strcmp(rev, temp) == 0) {
        return true;
    } else {
        return false;
    }
    
}
